MM-30596. Added preferences.cfg reference article.

rachelbritton opened the pull request:
Created reference article describing the preference file changes.

Also made some grammatical/terminology fixes on pages affected by the preference file updates, @mmOrvar.

rachelbritton committed:
MM-30596. Added preferences.cfg reference article.

Created reference article describing the preference file changes.

Also made some grammatical/terminology fixes on pages affected by the preference file updates.

mmOrvar commented on guides/implementation/webapp_fixed_central.rst:
```diff
@@ -7,7 +7,7 @@
 ===================================================================
 
-By default, the Micetro Web Interface and API only allow connecting to a single Micetro Central server, determined during the first login to Micetro after installation.
+By default, the Micetro ÙI and API only allow connecting to a single Micetro Central server, determined during the first login to Micetro after installation.
```
ÙI 👀

mmOrvar replied:
ÙI -> UI

rachelbritton committed:
MM-30596. Fixed typo.

Fixed UI typo.

mmOrvar commented on guides/management-console/external_auth.rst:
```diff
@@ -146,15 +146,15 @@ Micetro can authenticate using an external RADIUS server. In large installations
-To enable RADIUS authentication, you must add several properties to the Men&Mice Central configuration file preferences.cfg. This file is located in the data folder inside the Men&Mice Central data directory:
+To enable RADIUS authentication, you must add several properties to the Micetro Central configuration file preferences.cfg. This file is located in the data folder inside the Micetro Central data directory:
 
 * Windows: C:\\Program Files\\Men&Mice\\Central\\data
```
The double slashes here are a copy&paste artefact.

rachelbritton replied:
So there should be only one \ between each item?

mmOrvar replied:
There are more examples of this on other pages:
https://www.google.com/search?q=site%3Adocs.menandmice.com+%22%5C%5C%22

rachelbritton replied:
Looks like when this builds, the \\ formats the path as \ so no changes are needed here.

mmOrvar commented on guides/implementation/central_ha.rst:
```diff
@@ -21,13 +21,13 @@ Failover instances of Micetro Central can be configured to build a high availabi
 -----
 
-1. On the existing (or designated as *primary*) server running Central, edit the preferences file in ``/var/mmsuite/mmcentral/preferences.cfg``, adding
+1. On the existing (or designated as *primary*) server running Central, edit the preferences file in ``/var/mmsuite/mmcentral/preferences.cfg``, by adding the following to the end of the file:
```
to the end of the file -> "to the file" or "to the top of the file".

There are potentially problems with adding to the end of the file if sections are being used in the file.

mmOrvar replied:
to the end of the file -> to the top of the file

mmOrvar commented on guides/implementation/central_ha.rst:
```diff
@@ -90,13 +90,13 @@ Proceed to :ref:`install-controllers`.
 -------
 
-1. In the existing (or designated as *primary*) server running Central, edit the preferences file ``preferences.cfg``, adding
+1. In the existing (or designated as *primary*) server running Central, edit the preferences file ``preferences.cfg``, by adding the following to the end of the file:
```
Skip "end of the".

mmOrvar replied:
to the end of the file -> to the top of the file

mmOrvar commented on guides/implementation/central_postgresql.rst:
```diff
@@ -50,7 +50,7 @@ If the database connection fails, the service will not start and an error messag
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The preferences.cfg file located in the data directory contains, in addition to the fingerprint of Central (the "password" XML-tag), four other XML tags: database, databaseserver, databaseusername and databasepassword.
+The preferences.cfg file located in the data directory contains, in addition to the fingerprint of Central (the "password" XML-tag), four other XML tags: ``database``, ``databaseserver``, ``databaseusername`` and ``databasepassword``.
```
XML-tag -> preference
XML tags -> preferences
``database`` -> ``DatabaseType``
I would also use Pascal casing for ``DatabaseUsername`` and ``DatabasePassword`` for readability.

mmOrvar replied:
If you see XML and/or tag being used in regards to the preferences it should be changed.

mmOrvar commented on guides/reference/central_entra.rst:
```diff
@@ -42,7 +42,7 @@ This topic describes how to configure the Central server for multifactor authent
          * `mm_auth_cb.signature.zip <https://github.com/menandmice/micetro_docs/blob/latest/scripts/mm_auth_cb.signature.zip.zip>`_
          
-         For security reasons the script is signed and will not be run if there is not a matching signature file mm_auth_cb.signature in the same folder. 
+         For security reasons, the script is signed and will not be run if there is not a matching signature file mm_auth_cb.signature in the same folder. 
```
``mm_auth_cb.signature``

mmOrvar commented on guides/reference/changes_preference_file.rst:
```diff
@@ -0,0 +1,117 @@
+.. meta::
```
I was thinking that we would link to this page, that talks about the transformation from XML to INI, in the release notes, and then have the other page that only explains INI as part of the general documentation for v11.2 and following versions.

rachelbritton replied:
See revisions in new commit.

rachelbritton committed:
MM-30596. Revised config file pages.

Revised the two config file pages (in Installation and Reference Articles) based on feedback.

Fixed some grammar and terminology in affected pages.

mmOrvar commented on guides/admin-manual/failover_management_kea.rst:
```diff
@@ -107,7 +107,7 @@ Configuring the DHCP Agent
 The ``preferences.cfg`` file can be used to fine-tune the Kea High Availability setup within the DHCP agent.
 
-* ``KeaReplicateConfig``: If set to 0, Micetro will not replicate between Kea DHCP servers in a High Availability setup. The default value is **1**. Example: ``KeaReplicateConfig = false``
+* ``KeaReplicateConfig``: If set to 0, Micetro will not replicate between Kea DHCP servers in a High Availability setup. The default value is **1**. Example: ``KeaReplicateConfig = false``.
```
0 -> ``false``
1 -> ``true``

mmOrvar commented on guides/implementation/central_postgresql.rst:
```diff
@@ -62,7 +62,7 @@ The preferences.cfg file for standard user/password authentication should look l
   DatabasePassword = "password hash"
 
-For the Windows Authentication method, an example preferences.cfg file would look like  this (the databaseusername tag must be included and its value attribute should be an empty string):
+For the Windows Authentication method, an example preferences.cfg file would look like  this (the ``databaseusername`` tag must be included and its value attribute should be an empty string):
```
tag -> preference

mmOrvar commented on guides/implementation/central_postgresql.rst:
```diff
@@ -50,7 +50,7 @@ If the database connection fails, the service will not start and an error messag
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The preferences.cfg file located in the data directory contains, in addition to the fingerprint of Central (the "password" XML-tag), four other XML tags: database, databaseserver, databaseusername and databasepassword.
+The preferences.cfg file located in the data directory contains, in addition to the fingerprint of Central (the "password" XML-tag), four other XML tags: ``database``, ``databaseserver``, ``databaseusername`` and ``databasepassword``.
```
XML-tag -> preference
XML-tags -> preferences

rachelbritton committed:
MM-30596. Removed all XML tag references.

Revised based on feedback and changed all references to XML tags to "preferences".
